project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module

```diff
@@ -564,6 +564,26 @@ function c4m_message_format_event_user_full_name(Message $message) {
 }
 
 /**
+ * Returns a link to event user profile.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Returns a link to new member public profile.
+ */
+function c4m_message_format_event_user_url(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $uid = $wrapper->field_related_user->value(array('identifier' => TRUE));
```

Comment:
@ordavidil 
This will require creating and adding a new field to messages.
Being explicit here, means we need to add new field for every type of user we need to store with message.
For example, at message 1 we store 'event user', at message 2 we store 'new member and at message 3 we store 'group owner'.
Currently, we use field_related_user for all of them, while field label represents what's this field usage: 
![selection_036](https://cloud.githubusercontent.com/assets/17882245/17521049/948198d8-5e5a-11e6-9153-234199128e4c.png)
Being explicit requires 2 additional fields, and multiple new functions to resolve tokens from them.